### PR TITLE
Parser Tests: Check all error messages on parse failure

### PR DESCRIPTION
**Pull request by ajor**
Most of the errors look good. Some have some minor issues but I'm just locking in the existing behaviour with this PR.

### CODE REVIEW

File: tests/parser.cpp [hunk 1]
```diff
@@ -22,12 +22,12 @@ void test_parse_failure(BPFtrace &bpftrace,
   if (expected_error.data()) {
     if (!expected_error.empty() && expected_error[0] == '\n')
       expected_error.remove_prefix(1); // Remove initial '\n'
-    EXPECT_EQ(out.str(), expected_error);
+    EXPECT_EQ(expected_error, out.str());
```
Comment (ajor) [inline tests/parser.cpp:25]: Flipped the expected and actual results to make GTest give nicer error messages

File: tests/parser.cpp [hunk 2]
```diff
@@ -36,7 +36,7 @@ void test_parse_failure(std::string_view input,
 void test(BPFtrace &bpftrace, std::string_view input, std::string_view expected)
 {
   Driver driver(bpftrace);
-  EXPECT_EQ(driver.parse_str(input), 0);
+  ASSERT_EQ(driver.parse_str(input), 0);
```
Comment (ajor) [inline tests/parser.cpp:39]: If this fails then `driver.root` will be a null pointer and `Printer` will crash below trying to display it

File: tests/parser.cpp [hunk 3]
```diff
@@ -277,11 +281,36 @@ TEST(Parser, positional_param_attachpoint)
   int: 1
 )PROG");
 
-  test_parse_failure(bpftrace, R"PROG(uprobe:$1a" { 1 })PROG");
-  test_parse_failure(bpftrace, R"PROG(uprobe:$a" { 1 })PROG");
-  test_parse_failure(bpftrace, R"PROG(uprobe:$-1" { 1 })PROG");
+  // TODO what are these tests meant to be checking for?
```
Comment (ajor) [inline tests/parser.cpp:284]: These tests confused me. Can anyone help to explain why there are trailing, unmatched quotes (`"`) in each of them?

Do they serve a purpose or could they be a mistake?

Comment (danobi) [reply tests/parser.cpp:283]: I git-blamed it and it looks like I added it. Probably a typo. Good that you're adding the expected error messages :)

Comment (ajor) [reply tests/parser.cpp]: Cool, removed the trailing quotes!

File: tests/parser.cpp [hunk 4]
```diff
@@ -2131,17 +2422,54 @@ uprobe:asdf:Stream {} tracepoint:only_one_arg {}
...
+config = { BPFTRACE_STACK_MODE=perf } i:s:1 { BPFTRACE_MAX_PROBES=2; exit(); }
+                                              ~~~~~~~~~~~~~~~~~~~~
+)");
+  test_parse_failure("config { BPFTRACE_STACK_MODE=perf } i:s:1 { exit(); }",
+                     R"(
+stdin:1:8-9: ERROR: syntax error, unexpected {, expecting =
+config { BPFTRACE_STACK_MODE=perf } i:s:1 { exit(); }
+       ~
+)");
+  test_parse_failure("BPFTRACE_STACK_MODE=perf; i:s:1 { exit(); }", R"(
```
Comment (jordalgo) [inline tests/parser.cpp:2468]: nit: I know the rest of the file doesn't adhere to this but I like a little breathing room (line break) between the `test_parse_failure`s 🌬️

**Comment by jordalgo** [comment]
@ajor is an error message extraordinaire

**Review by jordalgo — APPROVED** [review]

File: tests/parser.cpp (hunk 4 above)
Comment (ajor) [reply tests/parser.cpp:2521]: Good point, definitely easier to read this way

**Pull request merged**